code/components/jomjol_tfliteclass/CTfLiteClass.cpp

```diff
@@ -305,17 +306,17 @@ CTfLiteClass::CTfLiteClass()
     this->input = nullptr;
     this->output = nullptr;  
     this->kTensorArenaSize = 800 * 1024;   /// according to testfile: 108000 - so far 600;; 2021-09-11: 200 * 1024
-    this->tensor_arena = (uint8_t*)GET_MEMORY(kTensorArenaSize);
+    this->tensor_arena = (uint8_t*)malloc_psram_heap(std::string(TAG) + "->tensor_arena", kTensorArenaSize, MALLOC_CAP_SPIRAM);
 }
 
 
 CTfLiteClass::~CTfLiteClass()
 {
-  free(modelfile);
-
-  free(this->tensor_arena);
   delete this->interpreter;
   delete this->error_reporter;
+
+  free_psram_heap(std::string(TAG) + "->modelfile", modelfile);
```

Comment:
There is no difference if you call free or caps_malloc_free even memory is allocated in PSRAM.

https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/mem_alloc.html
Because malloc() uses the capabilities-based allocation system, memory allocated using [heap_caps_malloc()](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/mem_alloc.html#_CPPv416heap_caps_malloc6size_t8uint32_t) can be freed by calling the standard free() function.

Reply:
yes, that is what i also would expect, but my experience below (https://github.com/jomjol/AI-on-the-edge-device/pull/2206#issuecomment-1474788761) seems to prove different!